docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+    -   `minted`: The amount of cycles created from the ICP payment
+    -   `balance`: The new balance of the target cycles ledger subaccount
+-   **Err**: Indicates why the minting failed (e.g., incorrect memo, controller mismatch, payment already processed, etc.).
+    For information about retry semantics, error handling, and caching behavior, see [Notify Methods – Shared Behavior](#notify-methods--shared-behavior-retry-error-handling-and-caching).
+
+## Notify Methods – Shared Behavior (Retry, Error Handling, and Caching)
+
+All `notify_*` methods (such as `notify_create_canister` and `notify_top_up`) follow the same core semantics:
+
+-   Before processing any specific operation (like create or top-up), the CMC checks the memo of the incoming ICP transfer identified by `block_index`. If the memo (either the legacy `u64` memo or the ICRC-1 `blob` memo interpreted as a little-endian `u64`) does **not** match one of the recognized operation memos (`1095062083` for create, `1347768404` for top-up, `1414416717` for mint), the CMC attempts to automatically refund the ICP amount (minus the standard ledger transfer fee) back to the sender's account (`from` account in the transfer). If the refund succeeds or fails terminally, the call will return a `NotifyError::Refunded` or `NotifyError::InvalidTransaction` respectively. If the refund attempt fails transiently (e.g., ledger unavailable), the status is cleared, allowing a retry.
+-   If a concurrent `notify_*` call is already processing for the same transfer, the new call returns an error indicating that the operation is in progress (`NotifyError::Processing`).
+-   If a previous `notify_*` call has already successfully processed a transfer at the same block height, the new call returns the same cached result (Ok or a terminal Err like `InvalidTransaction` or `Refunded`).
+-   To support safe retries, the result of a successful call or a terminal error is **cached** for a bounded (but generous) number of blocks (`MAX_NOTIFY_HISTORY`). Calls referring to blocks older than the cache window will fail with `NotifyError::TransactionTooOld`.
+
+These semantics ensure that all `notify_*` methods are idempotent. Clients can safely retry using the original parameters (memo, block height, destination account).
+
+### Error Handling and Refunds (`NotifyError`)
+
+Calls to `notify_*` methods return a `variant { Ok: ...; Err: NotifyError }`. The `NotifyError` type indicates why an operation failed:
+
+```candid
+type NotifyError = variant {
+  Refunded: record { reason : text; block_index : opt nat64 }; // Request was invalid; ICP (minus fees) was refunded (or attempted). `block_index` is the refund transaction index, if successful.
+  InvalidTransaction: text; // Transaction details (e.g., destination, memo) don't match the request, or it was already processed for a different operation.
+  Processing: null; // The specified block_index is currently being processed by another call. Retry shortly.
```

Comment:
"Retry shortly" is what we ask the caller who sees this error to do, right?
I was first not sure if this indicates "the canister will retry shortly", so maybe we can make this more precise?